python/tvm/contrib/debugger/debug_result.py

```diff
@@ -0,0 +1,147 @@
+"""Graph debug results dumping class."""
+import os
+import json
+import numpy as np
+
+GRAPH_DUMP_FILE_NAME = '_tvmdbg_graph_dump.json'
+
+class DebugResult():
+    """Graph debug data module.
+
+    Data dump module manage all the debug data formatting.
+    Output data and input graphs are formatted and the dump to files.
+    Frontend read these data and graph for visualization.
+
+    Parameters
+    ----------
+    graph_json : str
+        The graph to be deployed in json format output by nnvm graph. The graph can only contain
+        one operator(tvm_op) that points to the name of PackedFunc in the libmod.
+
+    dump_path : str
+        Output data path is read/provided from frontend
+    """
+
+    def __init__(self, graph_json, dump_path):
+        self._dump_path = dump_path
+        self._time_list = []
+        self._parse_graph(graph_json)
+        # dump the json information
+        self.dump_graph_json(graph_json)
+
+    def _parse_graph(self, graph_json):
+        """Parse and extract the NNVM graph and update the nodes, shapes and dltype.
+
+        Parameters
+        ----------
+        graph_json : str or graph class
+           The graph to be deployed in json format output by nnvm graph.
+        """
+        json_obj = json.loads(graph_json)
+        self._nodes_list = json_obj['nodes']
+        self._shapes_list = json_obj['attrs']['shape']
+        self._dltype_list = json_obj['attrs']['dltype']
+        self._update_graph_json()
+
+    def _update_graph_json(self):
+        """update the nodes_list with name, shape and data type,
+        for temporarily storing the output.
+
+        Parameters
+        ----------
+        None
+        """
+
+        nodes_len = len(self._nodes_list)
+        for i in range(nodes_len):
+            node = self._nodes_list[i]
+            input_list = []
+            for input_node in node['inputs']:
+                input_list.append(self._nodes_list[input_node[0]]['name'])
+            node['inputs'] = input_list
+            dltype = str("type: " + self._dltype_list[1][i])
+            if 'attrs' not in node:
+                node['attrs'] = {}
+                node['op'] = "param"
+            else:
+                node['op'] = node['attrs']['func_name']
+            node['name'] = node['name'].replace("/", "_")
+            node['attrs'].update({"T": dltype})
+            node['shape'] = self._shapes_list[1][i]
+
+    def get_graph_nodes(self):
+        """Return the nodes list
+        """
+        return self._nodes_list
+
+    def get_graph_node_shapes(self):
+        """Return the nodes shapes list
+        """
+        return self._shapes_list
+
+    def get_graph_node_dltypes(self):
```

Comment:
feel it is better to name it 'dtypes', keep the name convention the same across the project.